lightning-block-sync/src/http.rs

```diff
@@ -348,9 +348,11 @@ impl HttpClient {
 
 		if !status.is_ok() {
 			// TODO: Handle 3xx redirection responses.
-			let error_details = match contents.is_ascii() {
-				true => String::from_utf8_lossy(&contents).to_string(),
-				false => "binary".to_string()
+			let error_details = match String::from(contents) {
+				// Check that the string is all-ASCII with no control characters before returning
+				// it.
+				Ok(s) if s.as_bytes().all(|c| c <= 0x7e && c >= 0x20) => s,
```

Comment:
Could use `is_ascii` and `!is_ascii_control` on `u8` data before converting.

Reply:
Ah! I spent an embarrassing amount of time the other day looking for that, but I guess mostly annoyed that that isn't exposed on `str` and only `u8`.